### DISCUSSION TITLE
Stream VODs

### DISCUSSION THREAD

**Discussion by compufox** [owncast/owncast#652] [General Owncast discussion]
Is there any interest/work being done on having owncast save and serve old streams, like twitch VODs?

I feel like this would be a perfect use case for S3 backend storage as well. 

Just wanted to throw this out to the community!

**Comment by gabek** [comment]
Yup, the work is scheduled!

**Reply by compufox** [reply]
Awesome! Sorry I posted this before scanning the open issues. Probably a bad move :sweat_smile:

**Reply by gabek** [reply]
haha it's no problem.  Here's our roadmap if you're interested: https://github.com/owncast/owncast/milestones

**Reply by compufox** [reply]
Thanks!!

**Comment by taintedcypher** [comment]
Just had this idea - Don't know if other stream platforms do it but...
What is along with the video, the chat messages were also saved, and we'd see the chat playback in real time as actual chat messages in the ui, when watching the vod?
The chat data could also be a source of content to be searched, in order to find a video, maybe.

**Reply by gabek** [reply]
With how vod/replay functionality is currently built, this is possible. It just requires keeping chat history forever, something that is currently purposefully wiped pretty often on purpose. I'm not sure I'm in favor of people being able to search the entire chat history, but the sync'ed chat is doable with the architecture.

**Reply by taintedcypher** [reply]
>  I'm not sure I'm in favor of people being able to search the entire chat history

I'm not sure either :)

**Reply by gabek** [reply]
It opens up a privacy conversation. Do you want what you said to be able to be replayed by strangers? You think you're just chatting with your friends and then your coworker watches the stream later? I'm not sure if I'd want that. In fact, I'm pretty sure I wouldn't, personally.

**Comment by ReachableCEO** [comment]
Does OwnCast support saving streams to VOD yet?

**Reply by gabek** [reply]
Yet? https://owncast.online/roadmap

**Reply by InfernalError** [reply]
So milestone v0.4.0?

Or are you implying it's already been implemented? I was looking into this as well and I couldn't find anything in the Admin page nor the documentation that mentions this.

**Reply by gabek** [reply]
Both. It's a long way away, but yes, [the bulk of the technical work has been done](https://github.com/owncast/owncast/pull/3395). However, all of the UI, management of replays and clips, design, etc etc etc will be a lot of work. And nothing has even been thought about there.

**Reply by gabek** [reply]
I questioned "Yet" because I dislike when people assume just because they want something it should be done. It's like me asking you "Did you give me $1000 yet?"

**Reply by InfernalError** [reply]
Good to know. I too am excited for this feature, and good work so far.

I wouldn't take the "yet?" too personally. I too wound up here after spinning up Owncast and wondering if it had VOD capability. I think it's just the fact that you said the work was scheduled 4 years ago, and the only links to the progress/scheduled implementation are the roadmap page, which I too had difficulty parsing, because the verbiage of that milestone refers to replaying clips of previous streams, which is a bit different and more complex than the baseline of what a VOD is which is just a previously recorded stream unclipped in its entirety. So it's like, is serving VODs implemented but clips and associated management aren't? Or is the whole shebang still in development?

It looks like given the status of that pull request that the answer is currently the latter. Hopefully this will at least clear it up for anyone else who shows up with that question.
